keras/layers/dense_attention_test.py

```diff
@@ -203,6 +203,30 @@ def test_calculate_scores_multi_dim(self):
     expected = np.array([[[7.64, 12.24, 16.84], [14.24, 22.84, 31.44]]],
                         dtype=np.float32)
     self.assertAllClose(expected, actual)
+ 
+  def test_calculate_scores_multi_dim_concat(self):
+    # Query tensor of shape [1, 2, 4]
+    q = np.array([[[1., 1.1, 1.2, 1.3], [2., 2.1, 2.2, 2.3]]], dtype=np.float32)
+    # Key tensor of shape [1, 3, 4]
+    k = np.array(
+        [[[1.5, 1.6, 1.7, 1.8], [2.5, 2.6, 2.7, 2.8], [3.5, 3.6, 3.7, 3.8]]],
+        dtype=np.float32)
+    attention_layer = dense_attention.Attention(score_mode='concat')
+    attention_layer.build(input_shape=([1, 2, 4], [1, 3, 4]))
+    actual = attention_layer._calculate_scores(query=q, key=k)
+    attention_layer.attention_v = 1
```

Comment:
I notice that it isn't clear what `attention_v` means. Can you rename it to a fully spelled out variable name?

Reply:
Ah, so in `concat` score is the product of a learnable parameter (here `attention_v`) and the hyperbolic tangent of the concatenation of the query and key vectors. 

![tiQkz](https://user-images.githubusercontent.com/56781123/149321433-4651d4de-cb4f-441c-8b57-55780e56a747.png)
v right here is what I am calling `attention_v` and as usual Wa is the scaling parameter and ht and hs are query and key respcetively. 
A pytorch implementation of the same can be found [here](https://github.com/spro/practical-pytorch/blob/master/seq2seq-translation/seq2seq-translation.ipynb) in code block 11.

Reply:
Rename it to something that isn't an abbreviation, like `attention_weights`. See the Keras API style guide comments on naming: https://github.com/keras-team/governance/blob/master/keras_api_design_guidelines.md#carefully-weigh-whether-a-new-feature-should-be-included

Reply:
Renamed `attention_v` to `concat_score_weight`. Does that work?